spec/unit/network/http/connection_spec.rb

```diff
@@ -176,9 +178,21 @@ def expects_limit_exceeded(conn)
       expect(result.code).to eq(503)
     end
 
+    it "should close the connection before sleeping" do
+      allow(http).to receive(:request).and_return(httpunavailable, httpok)
+
+      pool = Puppet.lookup(:http_pool)
+      expect(pool).to receive(:with_connection).with(site, anything).and_yield(http)
+
+      expect(http).to receive(:finish)
+
+      result = subject.get('/foo')
+    end
```

Comment:
I'm not super happy with this test. But it feels like I'd need to rewrite most of the tests in this file in order to make it better.....